pkg/nuclide/file-tree/lib/FileTreeController.js

```diff
@@ -741,16 +747,62 @@ class FileTreeController {
       onClose: () => this._closeDialog(),
       shouldSelectBasename: true,
     };
-    this._openDialog(<FileDialogComponent {...props} />);
+    this._openDialog(props);
   }
 
-  _openDialog(component: ReactElement): void {
+  openDuplicateDialog(): void {
+    var selection = this._getSelectedEntryAndDirectoryAndRoot();
+    if (!selection || selection.entry.isDirectory()) {
+      return;
+    }
+
+    var message = (
+      <div>
+        <div>Enter the new name for the file in the root:</div>
+        <div>{path.normalize(selection.root.getPath() + '/')}</div>
+      </div>
+    );
+
+    var {entry, root} = selection;
+
+    var props = {
+      rootDirectory: root,
+      initialEntry: entry,
+      message,
+      onConfirm: (rootDirectory, relativeFilePath) => {
+        var file = rootDirectory.getFile(relativeFilePath);
+        file.create().then(async (createdSuccessfully) => {
```

Comment:
This looks great, but let's stick to all async/await rather than switching between that and promises. I also reworded the detail message to remove the "We"; I find it eerie when a [computer](https://en.wikipedia.org/wiki/HAL_9000) responds like that :

```js
onConfirm: (rootDirectory, relativeFilePath) => {
  var file = rootDirectory.getFile(relativeFilePath);
  var createdSuccessfully = await file.create();
  if (createdSuccessfully) {
    await entry.read().then(text => file.write(text));
    this._reloadDirectory(entry.getParent());
    atom.workspace.open(file.getPath());
  } else {
    atom.notifications.addWarning(
      "Failed to duplicate file",
      {
        dismissable: true,
        detail: `There was a problem duplicating the file. Please check if the file "${file.getBaseName()}" already exists.`
      }
    );
  }
},
```